### ISSUE TITLE
Occasionally: CompilationError at GET / Showing console output -> empty

### ISSUE THREAD

**Issue by benjamin79** [phoenixframework/phoenix#1575]
Sometimes i get `CompilationError at GET / Showing console output` on different occasions. I can´t reproduce it. And it happens not very often. The server console shows nothing. Like it didn´t receive anything. Also the Browser output is empty. Today it happend after i click submit after filling out login fields (example from book). Doing the same a second time and it works. The error is not bound to a particular thing to raise.

How can i debug this error?

### Environment

* Elixir version (elixir -v): 1.2.2
* Phoenix version (mix deps): 1.14
* NodeJS version (node -v):  5.5.0
* NPM version (npm -v): 3.3.12
* Operating system: Windows 10

**Comment by chrismccord** [comment]
Does this happen during active code edits? Are you using anything like a shared volume to edit code with a VM? Do you have an editor plugin that does syntax checking that could be compiling code behind the scenes?

**Comment by benjamin79** [comment]
> Does this happen during active code edits?

No, only when i click a link or one time even on first page load without any edits before.

> Are you using anything like a shared volume to edit code with a VM?

No.Directly on hdd.

> 
Do you have an editor plugin that does syntax checking that could be compiling code behind the scenes?

I use Sublime Text 3 Build 3103 with Elixir, Elm Language Support and Material Theme as installed packages.

**Comment by chrismccord** [comment]
Can you recreate this, occasionally, on a fresh project?

**Comment by benjamin79** [comment]
I don´t have time at the moment. It may take several hours. But i will try to recreate it as soon as i can.

**Comment by benjamin79** [comment]
I don´t know if it has anything to do t´with the error. But i noticed, that sometime on a click (link doesnßt matter)  the css is gone. I will start to build a new app today and let you know if it happens again.

**Comment by josevalim** [comment]
I am afraid what is happening is that we refresh static assets by copying them over and over again because symlinks do not work on Windows. @benjamin79 if you do to _build/dev/lib/YOUR_APP_NAME/priv is it a link or a regular directory?

**Comment by benjamin79** [comment]
@josevalim directory, css is also there.

**Comment by benjamin79** [comment]
@chrismccord Yes, it happend even on a fresh project. This time on phoenix server start. So no click was involved. It never happend under osx.

**Comment by chrismccord** [comment]
Paging @OnorioCatenacci , do you have any time to help us diagnose and work on a potential fix for this? Thanks!

**Comment by OnorioCatenacci** [comment]
Hi Chris,

I think I can spare a few cycles.  I will look at it  tomorrow.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by OnorioCatenacci** [comment]
Hey, for whatever it's worth (and I don't know if it would help with the issue at all) you can create symlinks on Windows.  It's not a well-known feature but it is do-able.

http://www.howtogeek.com/howto/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/

Again I'll see what I can do about reproducing the behavior when I get a chance--probably tonight.  

@benjamin79 You said this on your original bug report "Today it happend [sic] after i click submit after filling out login fields (example from book)." Which book were you working the example from?  If I can work the same example, it may make it easier for me to reproduce the behavior.  If you could post the exact book and a page number that would be a help.

I'll post back as soon as I've got a chance to look at this.

**Comment by benjamin79** [comment]
@OnorioCatenacci thanks for looking into it. Page 89 from Programming Phoenix. But it happens on other links, submits, refresh as well. But it can take hours between. It seems to be connected with code changes. The same code works in osx without any error.

**Comment by OnorioCatenacci** [comment]
Ok, I'll see if I can reproduce the behavior or not.  Luckily I have a
clean image of Win10 on a VM and I've got an OSX box so I should be able to
do the same tests as you've done.

On Fri, Mar 18, 2016 at 12:15 PM, benjamin79 <notifications@github.com>
wrote:

> @OnorioCatenacci <https://github.com/OnorioCatenacci> thanks for looking
> into it. Page 89 from Programming Phoenix. But it happens on other links,
> submits, refresh as well. But it can take hours between. It seems to be
> connected with code changes. The same code works in osx without any error.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-198431078>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by josevalim** [comment]
@benjamin79 can you update your app to github so it is easier for @OnorioCatenacci to test it out? :)

**Comment by OnorioCatenacci** [comment]
Thanks Jose--I had forgotten about asking about that!

--
Onorio


On Fri, Mar 18, 2016 at 1:02 PM, José Valim <notifications@github.com>
wrote:

> @benjamin79 <https://github.com/benjamin79> can you update your app to
> github so it is easier for @OnorioCatenacci
> <https://github.com/OnorioCatenacci> to test it out? :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-198454868>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by benjamin79** [comment]
Here it is: [https://github.com/benjamin79/phoenix_windows_error](https://github.com/benjamin79/phoenix_windows_error)

I can´t upload the one with the login. But the same happens with this one as well. This one contains the basic phoenix app with only 3 models added. No routes / links. Some (about 1 of 50 ) changes in code can trigger the described error on refresh.

I´m by far no expert in this language, but it doens´t seem to be related to any particiluar code.

**Comment by OnorioCatenacci** [comment]
HI Benjamin,

Just wanted to double check a couple of details:

1.) Can you try Elixir 1.2.3 and see if this error occurs there too?

2.) Which version of Erlang is present?  I'm going to see if I can reproduce the problem with 18.3 (which is the most recent version) but of course if I can't I'll drop back to 1.2.2 and whichever version of Erlang you've got.

**Comment by benjamin79** [comment]
Same with Elixir 1.2.3. No code change was needed to trigger the error this time.  Trying Erlang 18.3 instead 18.2 now.

**Comment by OnorioCatenacci** [comment]
Did Erlang 18.3 have any impact on the issue Benjamin?  I'd be surprised if
it did but I did want to double check.  Thanks for testing 1.2.3 for rme
too!

--
Onorio


On Sat, Mar 19, 2016 at 1:29 PM, benjamin79 <notifications@github.com>
wrote:

> Same with Elixir 1.2.3. No code change was needed to trigger the error
> this time. Trying Erlang 18.3 instead 18.2 now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-198752944>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by OnorioCatenacci** [comment]
I haven't forgotten this guys--was just trying to get the work I'd been doing on directory names on Windows to a point that it was ready to share with others.  Now that that's done I'll see if I can figure out what's going on here.

**Comment by benjamin79** [comment]
@OnorioCatenacci No, Erlang 18.3 doesn´t make a difference. Thanks for your time looking into it.

**Comment by OnorioCatenacci** [comment]
Sorry Benjamin--did you mention which browser you're testing with?  I don't
recall if you mentioned it or not. I'm just about ready to start testing
(sorry for the delay) and I want to make sure I'm testing with the right
tools as well.

--
Onorio


On Tue, Mar 22, 2016 at 2:58 AM, benjamin79 <notifications@github.com>
wrote:

> @OnorioCatenacci <https://github.com/OnorioCatenacci> No, Erlang 18.3
> doesn´t make a difference. Thanks for your time looking into it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-199667513>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by OnorioCatenacci** [comment]
Ok, I've got everything installed and I'm trying to run your app but it doesn't look right.  All I see is this:

![image](https://cloud.githubusercontent.com/assets/153365/13958505/b5d5ec5c-f026-11e5-9b03-53f7a658c25b.png)

What am I missing here?

**Comment by benjamin79** [comment]
I´ve used chrome 49/50. But i will try firefox as well. I didn´t thought that the browser might be an issue. It seems you have a zoom. On the right side should be a menu. But that is not doing anything. I thought i put up a basic app (only 3 little models). So the chances that some code leads to the error are less. With this app the error appears on simple auto refresh. For example after a code refresh.

I´ve a test app for testing different things. This one shows the error as well. I will upload this as soon as i can.

The error is hard to catch, cause there doesn´t seem to be any logic behind. It happens on different occasions and not very often. Sometimes 2 times in 5 minutes. Then 2/3 hours without any.

@chrismccord  Is there any debug mode i can activate to log what happens behind the scene? `CompilationError at GET / Showing console output` - What has to happen that this error is shows, but no console output?

**Comment by benjamin79** [comment]
@OnorioCatenacci Here you can find the test app [https://github.com/benjamin79/test_windows_error](url) . Besides the CompilationError you can see in the console that app.js and app.css are sometimes not found just by clicking around in the app. 

Another thing i noticed is that the app is not slow on windows, but feels faster an linux / osx. But i don´t know if that has anything to do with the error.

**Comment by OnorioCatenacci** [comment]
Hi all,

Don't know if has any bearing on this issue but I was trying to create a
new test from scratch and when I selected "Y" from the Fetch and Install
dependencies prompt I got this:

[image: Inline image 1]

I've attached the npm-debug.log for your reference Chris.  I'll keep on
with this to see if I can repro the issue that Benjamin reported but I
wanted to make sure you saw this one too.

--
Onorio


On Tue, Mar 22, 2016 at 3:25 PM, benjamin79 <notifications@github.com>
wrote:

> I´ve used chrome 49/50. But i will try firefox as well. I didn´t thought
> that the browser might be an issue. It seems you have a zoom. On the right
> side should be a menu. But that is not doing anything. I thought i put up a
> basic app (only 3 little models). So the chances that some code leads to
> the error are less. With this app the error appears on simple auto refresh.
> For example after a code refresh.
>
> I´ve a test app for testing different things. This one shows the error as
> well. I will upload this as soon as i can.
>
> The error is hard to catch, cause there doesn´t seem to be any logic
> behind. It happens on different occasions and not very often. Sometimes 2
> times in 5 minutes. Then 2/3 hours without any.
>
> @chrismccord <https://github.com/chrismccord> Is there any debug mode i
> can activate to log what happens behind the scene? CompilationError at
> GET / Showing console output - What has to happen that this error is
> shows, but no console output?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-199974325>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by OnorioCatenacci** [comment]
Hi Benjamin,

I cloned that repo on my VM.  That's what I sent the screenshot of earlier
today.  I can't find that menu you mentioned at all.  That's why I was
going to just try one of my own.

Did you get that error about left-pad too when you installed?

--
Onorio


On Tue, Mar 22, 2016 at 5:46 PM, benjamin79 <notifications@github.com>
wrote:

> @OnorioCatenacci <https://github.com/OnorioCatenacci> Here you can find
> the test app https://github.com/benjamin79/test_windows_error <http://url>
> . Besides the CompilationError you can see in the console that app.js and
> app.css are sometimes not found just by clicking around in the app.
>
> Another thing i noticed is that the app is not slow on windows, but feels
> faster an linux / osx. But i don´t know if that has anything to do with the
> error.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-200044344>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by benjamin79** [comment]
Hi Onirio,

i don´t see your last screenshot in github. Just [image: Inline image 1]

Benjamin

**Comment by OnorioCatenacci** [comment]
Ok, so this is the app.js and app.css message that Benjamin was referring
to:

[image: Inline image 1]

These seem to be related to that message about left_pad.  When I start the
server with mix phoenix.server I get this:

[image: Inline image 3]


Benjamin did you get that error message about left-pad when you created
your website?

--
Onorio


On Tue, Mar 22, 2016 at 5:52 PM, Onorio Catenacci <Catenacci@ieee.org>
wrote:

> Hi Benjamin,
>
> I cloned that repo on my VM.  That's what I sent the screenshot of earlier
> today.  I can't find that menu you mentioned at all.  That's why I was
> going to just try one of my own.
>
> Did you get that error about left-pad too when you installed?
>
> --
> Onorio
>
>
> On Tue, Mar 22, 2016 at 5:46 PM, benjamin79 <notifications@github.com>
> wrote:
>
>> @OnorioCatenacci <https://github.com/OnorioCatenacci> Here you can find
>> the test app https://github.com/benjamin79/test_windows_error
>> <http://url> . Besides the CompilationError you can see in the console
>> that app.js and app.css are sometimes not found just by clicking around in
>> the app.
>>
>> Another thing i noticed is that the app is not slow on windows, but feels
>> faster an linux / osx. But i don´t know if that has anything to do with the
>> error.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly or view it on GitHub
>> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-200044344>
>>
>
>
>
> --
> Onorio Catenacci
>
> http://onor.io
> http://www.google.com/+OnorioCatenacci
>
>


-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by OnorioCatenacci** [comment]
Thanks Benjamin.  I'll manually attach the screenshots if I need to.

--
Onorio


On Tue, Mar 22, 2016 at 6:02 PM, benjamin79 <notifications@github.com>
wrote:

> Hi Onirio,
>
> i don´t see your last screenshot in github. Just [image: Inline image 1]
>
> Benjamin
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-200050994>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by OnorioCatenacci** [comment]
The png files are up there now but they're not inlined.  You will need to
click on the links to see them.

--
Onorio


On Tue, Mar 22, 2016 at 6:02 PM, benjamin79 <notifications@github.com>
wrote:

> Hi Onirio,
>
> i don´t see your last screenshot in github. Just [image: Inline image 1]
>
> Benjamin
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-200050994>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by benjamin79** [comment]
I downloaded the test app from github and did mix deps.get and npm install without an error.

**Comment by OnorioCatenacci** [comment]
That particular error (about npm install) was not on your test app; it was
on a brand new one that I created.  And it seems to have something to do
with me not having brunch installed first.  After I manually installed
brunch I created another test app on my VM and I did not get that error.

I saw this in the docs:

*A note about Brunch.io before we begin: Phoenix will use Brunch.io for
asset management by default. Brunch.io's dependencies are installed via the
node package manager, not mix. Phoenix will prompt us to install them at
the end of the mix phoenix.new task. If we say "no" at that point, and if
we don't install those dependencies later with npm install, our application
will raise errors when we try to start it, and our assets may not load
properly. If we don't want to use Brunch.io at all, we can simply pass
--no-brunch to mix phoenix.new.*

I just didn't get that it meant that brunch had to be installed _before_
that last step of the phoenix.new task.  I thought that that last step
would install brunch.io too.  Chris, that may be an area where the docs
need to be updated, possibly.  Or I may have missed that part too :)

So, I'm still working on your app to try to get a repro of the error you're
seeing.  I only tried creating my own because I wasn't having any luck with
yours.

--
Onorio


On Tue, Mar 22, 2016 at 6:35 PM, benjamin79 <notifications@github.com>
wrote:

> I downloaded the test app from github and did mix deps.get and npm install
> without an error.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-200063230>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by OnorioCatenacci** [comment]
Benjamin,

I can't get either of those demo apps you've shared working to the point
that I would be able to try to reproduce your issue.

When I try to run phoenix_windows_error, I get the screen shot I posted at
first.  There's no menu visible anywhere.  And when I inspect the page in
the Chrome console, I see this message:

Unable to join Object
reason: "unmatched topic"
__proto__: Object
Socket.js: 60

I'm assuming this has something to do with your actual application but I
can't tell.

When I attempt to run test_windows_error I get various issues that seem to
indicate that ecto isn't setup correctly.
For example, if I try to register a new user, I get a message:
ERROR: (undefined_table): relation "users" does not exist.

When I attempt to click the mitarbeller link from the menu, I get this
error:
ERROR: (undefined_table): relation "employees" does not exist.

So in spite of the fact that mix ecto.create reports that it's created the
database fine (twice, because I ran an ecto.drop once too to insure I was
getting that right), it still seems to be wrong.

Now I think I can get your database if you care to share it.  Check this:

http://www.postgresql.org/docs/9.1/static/backup-dump.html

But I'm at a bit of a loss to figure out what I can do to reproduce this
problem unless I can get something that's basically functional.  Or at
least functional enough that I don't have to deal with all these issues
before I even get to the step of trying to reproduce the error you've
reported.

Another thought that crosses my mind is this.  I have VMWare.  If you can
capture a VM image of the machine with the issue, I can put it on my VMware
and I should easily be able to recreate and study the problem that way.
Here's a post about creating a VM from a physical machine:

http://www.serverwatch.com/server-tutorials/how-to-convert-a-physical-computer-to-a-virtual-machine.html

Please let me know what you want to do.

-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by benjamin79** [comment]
Here is the backup file. Do you see the menu in test_windows_error? If not, what resolution do you use?
[backup_rumbl.zip](https://github.com/phoenixframework/phoenix/files/185162/backup_rumbl.zip)

**Comment by OnorioCatenacci** [comment]
Yes I  see the menu in test_windows_error.  I will extract that db and look
at this tomorrow.

**Comment by benjamin79** [comment]
login for the app:
username:tellsapfel
pw: 123456

**Comment by chrismccord** [comment]
Ignore the node/npm issues for now as they are related to a user unpublishing some popular packages (such as left-pad apparently) and it breaking the whole internet due to nodes tendendency for tiny modules :(

**Comment by OnorioCatenacci** [comment]
This did raise a question for me.  Benjamin can you add the version of
Brunch you have installed?

**Comment by benjamin79** [comment]
brunch@2.5.1

**Comment by OnorioCatenacci** [comment]
Ok, Benjamin--very dumb question but I'm not super familiar with Node.  How
do I check the version of brunch I've got present to insure I'm on the same
version as you?

--
Onorio


On Wed, Mar 23, 2016 at 12:49 PM, benjamin79 <notifications@github.com>
wrote:

> brunch@2.5.1
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-200435708>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by chrismccord** [comment]
Can we try to rule out node by stripping it and all asset compilation out and seeing if we can still recreate the original issue?

**Comment by OnorioCatenacci** [comment]
Sounds good to me Chris.  Probably be quicker and easier for Benjamin to
try that since I'm not at the point of having his repro code working just
yet.

For my edification though--how would I go about stripping out node and
asset compilation if I need to do so?  Just remove node from the machine?

On Wed, Mar 23, 2016 at 1:29 PM, Chris McCord <notifications@github.com>
wrote:

> Can we try to rule out node by stripping it and all asset compilation out
> and seeing if we can still recreate the original issue?
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by benjamin79** [comment]
@OnorioCatenacci `npm list` in terminal

@chrismccord "stripping it" - do you mean i should remove node from windows and all npm packages in the app folder?

**Comment by OnorioCatenacci** [comment]
Thanks Benjamin.



On Wed, Mar 23, 2016 at 3:31 PM, benjamin79 <notifications@github.com>
wrote:

> @OnorioCatenacci <https://github.com/OnorioCatenacci> npm list in terminal
>
> @chrismccord <https://github.com/chrismccord> "stripping it" - do you
> mean i should remove node from windows and all npm packages in the app
> folder?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/phoenixframework/phoenix/issues/1575#issuecomment-200508584>
>



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

**Comment by OnorioCatenacci** [comment]
Ok all:

1.) Version of Brunch for me is also 2.5.1

2.) I've got the app working on my VM now--test_windows_error.  i tried
adding a business, tried adding a contact and a few other things.  I know
you said the error is intermittent at best but can you think of what you
were doing when you ran into the error last?

3.) I don't want to strip out Node and asset recompilation until I can see
if I can reproduce this problem.  Is there anything in the log that I could
key off of that would show me that an asset recompiled?  Some string I
could look for?  I want to see if maybe I haven't triggered an asset
recompilation yet.

I am attaching the log that I captured in the Win10 command console for the
session I ran.



-- 
Onorio Catenacci

http://onor.io
http://www.google.com/+OnorioCatenacci

[info] Running Rumbl.Endpoint with Cowboy using http on port 4000
23 Mar 22:07:32 - info: compiling
23 Mar 22:07:34 - info: compiled 7 files into 2 files, copied 3 in 8.8 sec
[info] GET /
[debug] Processing by Rumbl.PageController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] Sent 200 in 263ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /sessions/new
[debug] Processing by Rumbl.SessionController.new/2
  Parameters: %{}
  Pipelines: [:browser]
[info] Sent 200 in 72ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] POST /sessions
[debug] Processing by Rumbl.SessionController.create/2
  Parameters: %{"_csrf_token" => "K0kGBBtbCSk5KBloZBZmAgJQKCU4NgAAi1tUL0OhSPtQWF6lr1CNyQ==", "_utf8" => "Γ£ô", "session" => %{"password" => "[FILTERED]", "username" => "telsapfel"}}
  Pipelines: [:browser]
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."username" = $1) ["telsapfel"] OK query=703.0ms queue=94.0ms
[info] Sent 500 in 1179ms
[error] #PID<0.404.0> running Rumbl.Endpoint terminated
Server: localhost:4000 (http)
Request: POST /sessions
** (exit) an exception was raised:
    ** (CondClauseError) no cond clause evaluated to a true value
        (rumbl) Rumbl.Auth.login_by_username_and_pass/4
        (rumbl) web/controllers/session_controller.ex:9: Rumbl.SessionController.create/2
        (rumbl) web/controllers/session_controller.ex:1: Rumbl.SessionController.action/2
        (rumbl) web/controllers/session_controller.ex:1: Rumbl.SessionController.phoenix_controller_pipeline/2
        (rumbl) lib/phoenix/router.ex:261: Rumbl.Router.dispatch/2
        (rumbl) web/router.ex:1: Rumbl.Router.do_call/2
        (rumbl) lib/rumbl/endpoint.ex:1: Rumbl.Endpoint.phoenix_pipeline/1
        (rumbl) lib/plug/debugger.ex:93: Rumbl.Endpoint."call (overridable 3)"/2
        (rumbl) lib/phoenix/endpoint/render_errors.ex:34: Rumbl.Endpoint.call/2
        (plug) lib/plug/adapters/cowboy/handler.ex:15: Plug.Adapters.Cowboy.Handler.upgrade/4
        (cowboy) src/cowboy_protocol.erl:442: :cowboy_protocol.execute/4
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] POST /sessions
[debug] Processing by Rumbl.SessionController.create/2
  Parameters: %{"_csrf_token" => "K0kGBBtbCSk5KBloZBZmAgJQKCU4NgAAi1tUL0OhSPtQWF6lr1CNyQ==", "_utf8" => "Γ£ô", "session" => %{"password" => "[FILTERED]", "username" => "tellsapfel"}}
  Pipelines: [:browser]
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."username" = $1) ["tellsapfel"] OK query=47.0ms
[info] Sent 302 in 46ms
[info] GET /
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.PageController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] Sent 200 in 59ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /companies
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.CompanyController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0) AS temp [] OK query=15.0ms
[info] SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0 LIMIT $1 OFFSET $2 [2, 0] OK query=0.0ms
[info] Sent 200 in 93ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /companies/new
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.CompanyController.new/2
  Parameters: %{}
  Pipelines: [:browser]
[info] Sent 200 in 37ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] POST /companies
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.CompanyController.create/2
  Parameters: %{"_csrf_token" => "LQgIPW8sIxVfFTlNciRmIwMXAxsWNgAAopzl8GeT5mTtAt6MsvhpWQ==", "_utf8" => "Γ£ô", "company" => %{"city" => "Rochester", "notes" => "", "phone" => "(248) 425-8634", "street" => "1425 Lantern Lane", "title" => "Bob's Business", "url" => "http://www.bobs.com", "zip" => "48306"}}
  Pipelines: [:browser]
[info] Sent 200 in 49ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] POST /companies
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.CompanyController.create/2
  Parameters: %{"_csrf_token" => "Kjs7PhECIzI+Pzx4VDwDIiADWiAvEAAAhCIoFiesTGQAglSLPb1Knw==", "_utf8" => "Γ£ô", "company" => %{"city" => "Rochester", "notes" => "Really a new business", "phone" => "(248) 425-8634", "street" => "1425 Lantern Lane", "title" => "Bob's Business", "url" => "http://www.bobs.com", "zip" => "48306"}}
  Pipelines: [:browser]
[info] BEGIN [] OK query=0.0ms
[info] INSERT INTO "companies" ("inserted_at", "updated_at", "category_id", "city", "notes", "phone", "street", "title", "url", "user_id", "zip") VALUES ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11) RETURNING "id" [{{2016, 3, 23}, {22, 9, 59, 0}}, {{2016, 3, 23}, {22, 9, 59, 0}}, nil, "Rochester", "Really a new business", "(248) 425-8634", "1425 Lantern Lane", "Bob's Business", "http://www.bobs.com", nil, "48306"] OK query=0.0ms
[info] COMMIT [] OK query=0.0ms
[info] Sent 302 in 130ms
[info] GET /companies
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.CompanyController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0) AS temp [] OK query=0.0ms
[info] SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0 LIMIT $1 OFFSET $2 [2, 0] OK query=0.0ms
[info] Sent 200 in 6ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /companies
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.CompanyController.index/2
  Parameters: %{"page" => "2"}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0) AS temp [] OK query=15.0ms
[info] SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0 LIMIT $1 OFFSET $2 [2, 2] OK query=0.0ms
[info] Sent 200 in 28ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /companies
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.CompanyController.index/2
  Parameters: %{"page" => "1"}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0) AS temp [] OK query=0.0ms
[info] SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0 LIMIT $1 OFFSET $2 [2, 0] OK query=0.0ms
[info] Sent 200 in 6ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /contacts
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.ContactController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."firstname", c0."lastname", c0."inserted_at", c0."updated_at" FROM "contacts" AS c0) AS temp [] OK query=16.0ms
[info] SELECT c0."id", c0."firstname", c0."lastname", c0."inserted_at", c0."updated_at" FROM "contacts" AS c0 LIMIT $1 OFFSET $2 [2, 0] OK query=0.0ms
[info] Sent 200 in 18ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /contacts/new
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=16.0ms
[debug] Processing by Rumbl.ContactController.new/2
  Parameters: %{}
  Pipelines: [:browser]
[info] Sent 200 in 4ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] POST /contacts
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.ContactController.create/2
  Parameters: %{"_csrf_token" => "EBMwCW4ELzImL1t9aWcEPio1JQQVAAAARkBX9oisLW6DZ7TPZTNoTg==", "_utf8" => "Γ£ô", "contact" => %{"firstname" => "David", "lastname" => "Malinowski"}}
  Pipelines: [:browser]
[info] INSERT INTO "contacts" ("inserted_at", "updated_at", "firstname", "lastname") VALUES ($1, $2, $3, $4) RETURNING "id" [{{2016, 3, 23}, {22, 10, 51, 0}}, {{2016, 3, 23}, {22, 10, 51, 0}}, "David", "Malinowski"] OK query=0.0ms
[info] Sent 302 in 10ms
[info] GET /contacts
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.ContactController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."firstname", c0."lastname", c0."inserted_at", c0."updated_at" FROM "contacts" AS c0) AS temp [] OK query=0.0ms
[info] SELECT c0."id", c0."firstname", c0."lastname", c0."inserted_at", c0."updated_at" FROM "contacts" AS c0 LIMIT $1 OFFSET $2 [2, 0] OK query=16.0ms
[info] Sent 200 in 7ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /contacts
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.ContactController.index/2
  Parameters: %{"page" => "2"}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."firstname", c0."lastname", c0."inserted_at", c0."updated_at" FROM "contacts" AS c0) AS temp [] OK query=0.0ms
[info] SELECT c0."id", c0."firstname", c0."lastname", c0."inserted_at", c0."updated_at" FROM "contacts" AS c0 LIMIT $1 OFFSET $2 [2, 2] OK query=0.0ms
[info] Sent 200 in 6ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /contacts
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.ContactController.index/2
  Parameters: %{"page" => "1"}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."firstname", c0."lastname", c0."inserted_at", c0."updated_at" FROM "contacts" AS c0) AS temp [] OK query=0.0ms
[info] SELECT c0."id", c0."firstname", c0."lastname", c0."inserted_at", c0."updated_at" FROM "contacts" AS c0 LIMIT $1 OFFSET $2 [2, 0] OK query=0.0ms
[info] Sent 200 in 6ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /videos
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.VideoController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] SELECT c0."name", c0."id" FROM "categories" AS c0 ORDER BY c0."name" [] OK query=0.0ms
"__________________________________________________"
[:url, :title, :category_id]
"__________________________________________________"
[info] SELECT count(*) FROM (SELECT v0."id", v0."url", v0."title", v0."description", v0."user_id", v0."category_id", v0."inserted_at", v0."updated_at" FROM "videos" AS v0 LEFT OUTER JOIN "categories" AS c1 ON v0."category_id" = c1."id") AS temp [] OK query=0.0ms
[info] SELECT v0."id", v0."url", v0."title", v0."description", v0."user_id", v0."category_id", v0."inserted_at", v0."updated_at" FROM "videos" AS v0 LEFT OUTER JOIN "categories" AS c1 ON v0."category_id" = c1."id" ORDER BY v0."category_id" DESC LIMIT $1 OFFSET $2 [2, 0] OK query=0.0ms
[info] SELECT c0."id", c0."name", c0."inserted_at", c0."updated_at" FROM "categories" AS c0 WHERE (c0."id" IN ($1,$2)) [4, 2] OK query=15.0ms
[info] Sent 200 in 403ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /videos/2
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.VideoController.show/2
  Parameters: %{"id" => "2"}
  Pipelines: [:browser]
[info] SELECT c0."name", c0."id" FROM "categories" AS c0 ORDER BY c0."name" [] OK query=0.0ms
[info] SELECT v0."id", v0."url", v0."title", v0."description", v0."user_id", v0."category_id", v0."inserted_at", v0."updated_at" FROM "videos" AS v0 WHERE (v0."user_id" IN ($1)) AND (v0."id" = $2) [1, 2] OK query=0.0ms
[info] Sent 200 in 25ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /videos/2/edit
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.VideoController.edit/2
  Parameters: %{"id" => "2"}
  Pipelines: [:browser]
[info] SELECT c0."name", c0."id" FROM "categories" AS c0 ORDER BY c0."name" [] OK query=0.0ms
[info] SELECT v0."id", v0."url", v0."title", v0."description", v0."user_id", v0."category_id", v0."inserted_at", v0."updated_at" FROM "videos" AS v0 WHERE (v0."user_id" IN ($1)) AND (v0."id" = $2) [1, 2] OK query=0.0ms
[info] Sent 200 in 7ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] SELECT c0."id", c0."name", c0."inserted_at", c0."updated_at" FROM "categories" AS c0 WHERE (c0."id" = $1) [2] OK query=15.0ms
[info] SELECT c0."id", c0."name", c0."inserted_at", c0."updated_at" FROM "categories" AS c0 WHERE (c0."id" = $1) [4] OK query=0.0ms
[info] POST /videos/2
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.VideoController.update/2
  Parameters: %{"_csrf_token" => "CywHHQISdyoYHzp8BhEELwAtAV0pJgAAITuLUy1krgWE5ATApLj6hA==", "_method" => "put", "_utf8" => "Γ£ô", "id" => "2", "video" => %{"category_id" => "4", "description" => "Comedy", "title" => "Groundhog Day", "url" => "aaa"}}
  Pipelines: [:browser]
[info] SELECT c0."name", c0."id" FROM "categories" AS c0 ORDER BY c0."name" [] OK query=0.0ms
[info] SELECT v0."id", v0."url", v0."title", v0."description", v0."user_id", v0."category_id", v0."inserted_at", v0."updated_at" FROM "videos" AS v0 WHERE (v0."user_id" IN ($1)) AND (v0."id" = $2) [1, 2] OK query=0.0ms queue=16.0ms
[info] Sent 302 in 9ms
[info] GET /videos/2
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=16.0ms
[debug] Processing by Rumbl.VideoController.show/2
  Parameters: %{"id" => "2"}
  Pipelines: [:browser]
[info] SELECT c0."name", c0."id" FROM "categories" AS c0 ORDER BY c0."name" [] OK query=0.0ms
[info] SELECT v0."id", v0."url", v0."title", v0."description", v0."user_id", v0."category_id", v0."inserted_at", v0."updated_at" FROM "videos" AS v0 WHERE (v0."user_id" IN ($1)) AND (v0."id" = $2) [1, 2] OK query=0.0ms
[info] Sent 200 in 6ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /videos
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.VideoController.index/2
  Parameters: %{}
  Pipelines: [:browser]
"__________________________________________________"
[info] SELECT c0."name", c0."id" FROM "categories" AS c0 ORDER BY c0."name" [] OK query=0.0ms
[:url, :title, :category_id]
"__________________________________________________"
[info] SELECT count(*) FROM (SELECT v0."id", v0."url", v0."title", v0."description", v0."user_id", v0."category_id", v0."inserted_at", v0."updated_at" FROM "videos" AS v0 LEFT OUTER JOIN "categories" AS c1 ON v0."category_id" = c1."id") AS temp [] OK query=0.0ms
[info] SELECT v0."id", v0."url", v0."title", v0."description", v0."user_id", v0."category_id", v0."inserted_at", v0."updated_at" FROM "videos" AS v0 LEFT OUTER JOIN "categories" AS c1 ON v0."category_id" = c1."id" ORDER BY v0."category_id" DESC LIMIT $1 OFFSET $2 [2, 0] OK query=0.0ms
[info] SELECT c0."id", c0."name", c0."inserted_at", c0."updated_at" FROM "categories" AS c0 WHERE (c0."id" IN ($1,$2)) [4, 2] OK query=0.0ms
[info] Sent 200 in 54ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /employees
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.EmployeeController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] SELECT e0."id", e0."firstname", e0."lastname", e0."email", e0."phone", e0."tags", e0."image", e0."salutation_id", e0."position_id", e0."status_id", e0."inserted_at", e0."updated_at" FROM "employees" AS e0 [] OK query=0.0ms
[info] SELECT s0."id", s0."name", s0."inserted_at", s0."updated_at" FROM "salutation" AS s0 WHERE (s0."id" IN ($1)) [1] OK query=0.0ms
[info] SELECT s0."id", s0."name", s0."inserted_at", s0."updated_at" FROM "status" AS s0 WHERE (s0."id" IN ($1)) [1] OK query=0.0ms
[info] Sent 200 in 11ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /employees/1/edit
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.EmployeeController.edit/2
  Parameters: %{"id" => "1"}
  Pipelines: [:browser]
[info] SELECT s0."name", s0."id" FROM "salutation" AS s0 [] OK query=0.0ms
[info] SELECT s0."name", s0."id" FROM "status" AS s0 ORDER BY s0."name" [] OK query=16.0ms
[info] SELECT e0."id", e0."firstname", e0."lastname", e0."email", e0."phone", e0."tags", e0."image", e0."salutation_id", e0."position_id", e0."status_id", e0."inserted_at", e0."updated_at" FROM "employees" AS e0 WHERE (e0."id" = $1) [1] OK query=0.0ms
[info] Sent 200 in 10ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] POST /sessions/1
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=0.0ms
[debug] Processing by Rumbl.SessionController.delete/2
  Parameters: %{"_csrf_token" => "Oi8IJ25bMBAHLwx9dTobKwUYOSlxNgAAxWzv90vQmWaDFjKEuyRB0Q==", "_method" => "delete", "id" => "1"}
  Pipelines: [:browser]
[info] Sent 302 in 2ms
[info] GET /
[debug] Processing by Rumbl.PageController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] Sent 200 in 0┬╡s
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /videos
[debug] Processing by Rumbl.VideoController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] Sent 302 in 2ms
[debug] Rumbl.VideoController halted in :authenticate_user/2
[info] GET /
[debug] Processing by Rumbl.PageController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] Sent 200 in 0┬╡s
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /users
[debug] Processing by Rumbl.UserController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 [] OK query=0.0ms
[info] Sent 200 in 2ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /users/1
[debug] Processing by Rumbl.UserController.show/2
  Parameters: %{"id" => "1"}
  Pipelines: [:browser]
[info] SELECT u0."id", u0."name", u0."username", u0."password_hash", u0."inserted_at", u0."updated_at" FROM "users" AS u0 WHERE (u0."id" = $1) [1] OK query=16.0ms
[info] Sent 200 in 2ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok
[info] GET /companies
[debug] Processing by Rumbl.CompanyController.index/2
  Parameters: %{}
  Pipelines: [:browser]
[info] SELECT count(*) FROM (SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0) AS temp [] OK query=15.0ms
[info] SELECT c0."id", c0."title", c0."url", c0."phone", c0."street", c0."zip", c0."city", c0."notes", c0."user_id", c0."category_id", c0."inserted_at", c0."updated_at" FROM "companies" AS c0 LIMIT $1 OFFSET $2 [2, 0] OK query=0.0ms
[info] Sent 200 in 5ms
[info] JOIN rooms:lobby to Rumbl.RoomChannel
  Transport:  Phoenix.Transports.WebSocket
  Parameters: %{}
[info] Replied rooms:lobby :ok

**Comment by benjamin79** [comment]
@OnorioCatenacci I would say mostly logins and then on clicking /videos or /employees. Less on refresh.
Did you develop an app yourself in windows? If not that might be the best way to experience it. Cause it can take 2-3 hours between.

**Comment by benjamin79** [comment]
I don´t know if it has anything to do with the error. But now i noticed the second time that (days between) i have to restart the server after doing a refresh in the browser by hand after a code change. The page would otherwise load forever. The server console doesn´t show any output. The browser network shows pending on the get request url (like http://localhost:4000/tasks or http://localhost:4000/videos).

**Comment by chrismccord** [comment]
@OnorioCatenacci thanks for taking the time to try to pinpoint this with @benjamin79 . Since we haven't gotten any closer to a resolution, I'm going to declare a loss on this one until we have more information to move forward. Thanks!

**Comment by benjamin79** [comment]
@chrismccord ok. Besides osx i`ve used linux (arch) the last days. Same code, also no error and faster. I won't use windows any time soon again.

**Comment by Jabolek** [comment]
Hey guys, I did encounter the same issue and it was due to project being inside Dropbox synced folder. After turning Dropbox sync off it went away.

**Comment by benjamin79** [comment]
I was using local folders.
